smack-extensions/src/main/java/org/jivesoftware/smackx/avatar/MetadataInfo.java

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ * Copyright 2017 Fernando Ramirez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.avatar;
+
+/**
+ * User Avatar metadata info model class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0084.html">XEP-0084: User
+ *      Avatar</a>
+ */
+public class MetadataInfo {
+
+    private final String id;
+    private final String url;
+    private final long bytes;
+    private final String type;
+    private final int height;
+    private final int width;
+
+    /**
+     * MetadataInfo constructor.
+     * 
+     * @param id
+     * @param url
+     * @param bytes
+     * @param type
+     * @param pixelsHeight
+     * @param pixelsWidth
+     */
+    public MetadataInfo(String id, String url, long bytes, String type, int pixelsHeight, int pixelsWidth) {
+        this.id = id;
+        this.url = url;
+        this.bytes = bytes;
+        this.type = type;
+        this.height = pixelsHeight;
```

Comment:
height and width are xs:unsingedByte, thus the constructor should ensure that it's value is within the range of [1,254] and otherwise throw an IllegalArgumentException.